### DISCUSSION TITLE
DevTools as browser extension

### DISCUSSION THREAD

**Discussion by kachkaev** [jotaijs/jotai-devtools#59] [Ideas]
👋 folks! A Jotai newbie here 👻

I just installed DevTools into my first Jotai project and saw a somewhat noticeable diff in my `yarn.lock`. Overall, this is fine because project sub-dependencies are not going to affect the production bundle. However, I wonder if it is feasible to make DevTools available as a browser extension, similar to [Apollo Client DevTools](https://chrome.google.com/webstore/detail/apollo-client-devtools/jdkknkkbebbapilgoeccciglkfbmbnfm) and [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd). I remember using those when they just got released and they were only available as project imports. The maintainers eventually released the extensions and thus helped lots of developers with keeping their `node_modules` small.

How feasible is a browser extension for Jotai DevTools?

Either way, thanks for working on this repo – the UX in the DevTools is quite smooth 💯

**Comment by arjunvegda** [comment]
Thank you! Glad to hear that you're enjoying the DevTools! 

A browser extension is not on the roadmap currently, however, I'm excited about the possibility of exploring it in the future. Perhaps after we flush out a couple more features on the current dev-tools. 😄 

What do you think? @dai-shi

**Reply by dai-shi** [reply]
Well, I personally don't have the bandwidth to follow those things, but why not.

FWIW, there was originally a project led by someone for extension. https://github.com/pmndrs/jotai/discussions/918

**Comment by arjunvegda** [comment]
Small update: I'm playing around with the https://developer.chrome.com/docs/web-platform/document-picture-in-picture/ API and it seems promising. It'd most likely require quite a bit of refactoring to get it working.


![image](https://github.com/jotaijs/jotai-devtools/assets/14841132/222b67da-5721-4d47-819e-358cbcde7c35)

**Comment by tpavlu** [comment]
If I get some bandwidth I might take a stab at this.
I was thinking we could just take the same approach redux does and require defining a `window.__JOTAI_STORE__` reference

And then just render these dev tools with the hide option removed in a dev tools panel for that store right?

**Reply by arjunvegda** [reply]
I'm not sure how redux uses a window property 🤔. I'd consider mounting an iFrame or a Shadow dom and communicate with it using [postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage)
